server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java

```diff
@@ -174,37 +174,6 @@ public FloatVectorValues getFloatVectorValues(String field) throws IOException {
             }
             return queryCancellation.isEnabled() ? new ExitableFloatVectorValues(vectorValues, queryCancellation) : vectorValues;
         }
-
-        @Override
-        public void searchNearestVectors(String field, float[] target, KnnCollector collector, Bits acceptDocs) throws IOException {
-            if (queryCancellation.isEnabled() == false) {
-                in.searchNearestVectors(field, target, collector, acceptDocs);
-                return;
-            }
-            // when acceptDocs is null due to no doc deleted, we will instantiate a new one that would
-            // match all docs to allow timeout checking.
-            final Bits updatedAcceptDocs = acceptDocs == null ? new Bits.MatchAllBits(maxDoc()) : acceptDocs;
```

Comment:
This is the only change that isn't so obvious I think. The `timeoutCheckingAcceptDocs` variable wasn't used. If you remove it, the check for `queryCancellation.isEnabled()` becomes redundant and the method collapses into what the parent method does.